### ISSUE TITLE
[BUG] After shutting down the cluster the kubeconfig becomes invalid

### ISSUE THREAD

**Issue by guangbochen** [harvester/harvester#1475]
**Describe the bug**

After shutting down the cluster last night and starting it today and logged in to the web UI and downloaded the KubeConfig
but it gives me the following error: `Unable to connect to the server: x509: certificate is valid for 10.52.0.126, 10.52.1.39, 10.53.92.110, not 172.18.0.20`

The kube-vip is 172.18.0.20 and it looks like the cert is valid for the flannel networks?
```
>$ for i in 1 2 3; do ssh "node$i" sudo ip -brief a show flannel.1; done
flannel.1        UNKNOWN        10.52.0.0/32 fe80::7c0d:a2ff:fe5c:7aed/64 
flannel.1        UNKNOWN        10.52.1.0/32 fe80::e433:b8ff:fe67:e069/64 
flannel.1        UNKNOWN        10.52.2.0/32 fe80::7859:16ff:fedf:45eb/64
The nodes have IP 172.18.0.21-23
```

**Expected behavior**
Running kubectl on the nodes works as expected.

**Environment:**
 - Harvester ISO version: v0.3.0
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):

**Additional context**

**Comment by guangbochen** [comment]
A temp workaround is to ssh login to one of the management node and run `kubectl annotate secrets -n cattle-system tls-rancher-internal listener.cattle.io/cn-172.18.0.20=172.18.0.20` will fix the issue.

**Comment by Jellyfrog** [comment]
Can confirm this issue, and that the workaround works.

**Comment by verejoel** [comment]
I also had a similar issue. However I did not restart the main management node - I joined a virtualized harvester node to the cluster. After this, I was unable to use kubectl until I performed the workaround given by @guangbochen

**Comment by TachunLin** [comment]
Verified **fixed** on master-655c251a-head (based on v1.0.0-rc1). Close this issue.

### Result
Shutdown all 3 harvester cluster nodes machines and start after hours, can download the KubeConfig without error. 
Can run kubectl on all nodes correctly too. 


### Test Information
* Test Environment: 3 nodes harvester on local kvm machines
* Harvester version: master-655c251a-head (12/7, based on v1.0.0-rc1)

### Environment Setup
1. ISO install 3 nodes harvester cluster 

### Verify Steps

1. Shutdown harvester node 3, wait for fully power off
2. Shutdown harvester node 2, wait for fully power off
3. Shutdown harvester node 1, wait for fully power off
4. Wait for more than hours or over night
5. Power on node 1 to console page until you see management url 
![image](https://user-images.githubusercontent.com/29251855/145156486-60507643-8a96-4b4a-862d-367c41665e6b.png)

6.  Power on node 2 to console page until you see management url 
7.  Power on node 3 to console page until you see management url 
8.  Confirm all nodes are in `Ready` status
9.  Access harvester dashboard
10. Go to setting and click `Download Kubeconfig`
![image](https://user-images.githubusercontent.com/29251855/145156633-feddec8c-e322-404f-9104-7c543824f884.png)
11. ssh to each node and run `kubectl get pods -A` 

### Additional Context

```
rancher@harvester-node01-1207:~> for i in 1 2 3; do ssh "harvester-node0$i-1207" sudo ip -brief a show flannel.1; done
Password: 
flannel.1        UNKNOWN        10.52.0.0/32 fe80::3099:c0ff:fe15:2fd8/64 
Password: 
flannel.1        UNKNOWN        10.52.1.0/32 fe80::9814:18ff:fe8d:41e1/64 
Password: 
flannel.1        UNKNOWN        10.52.2.0/32 fe80::36:92ff:fed0:9641/64
```
